cglib/src/main/java/net/sf/cglib/core/KeyFactory.java

```diff
@@ -84,14 +84,30 @@
         938313161, 1288102441, 1768288259  };
     
 
-    public static final Customizer CLASS_BY_NAME = new Customizer() {
+    public static final Customizer CLASS_BY_NAME = new Customizer2() {
         public void customize(CodeEmitter e, Type type) {
             if (type.equals(Constants.TYPE_CLASS)) {
                 e.invoke_virtual(Constants.TYPE_CLASS, GET_NAME);
             }
         }
+
+        public void customize(CodeEmitter e, int index, Type type) {
```

Comment:
I did not touch `Customizer` interface to avoid backward-incompatible changes for no reason.
Adding a parameter to existing method is not a compatible change if clients have their own implementations on `Customizer`